tests/Fixer/FunctionNotation/VoidReturnFixerTest.php

```diff
@@ -260,6 +259,31 @@ abstract protected function foo($param);
         yield [
             '<?php fn($a) => var_dump($a);',
         ];
+
+        $excluded = ['__clone', '__construct', '__debugInfo', '__destruct', '__isset', '__serialize', '__set_state', '__sleep', '__toString'];
+
+        foreach (self::provideMagicMethodsDefinitions() as $magicMethodsDefinition) {
+            $name = $magicMethodsDefinition[0];
+            $arguments = $magicMethodsDefinition[1] ?? 0;
+            $isStatic = $magicMethodsDefinition[2] ?? false;
+            $code = sprintf(
+                '<?php class Test { public%s function %s(%s)%%s {} }',
+                $isStatic ? ' static' : '',
+                $name,
+                implode(',', array_map(
+                    static fn (int $n): string => sprintf('$x%d', $n),
+                    array_keys(array_fill(0, $arguments ?? 0, true)),
+                ))
+            );
+
+            $input = sprintf($code, '');
+            $expected = sprintf($code, \in_array($name, $excluded, true) ? '' : ': void');
+
+            yield [
```

Comment:
recovered from original method description

```suggestion
            yield sprintf('Test if magic method %s is handled without causing syntax error', $name) => [
```